### DISCUSSION TITLE
IconLayer, icon transparency when alphaCutoff set to 0.0

### DISCUSSION THREAD

**Discussion by mplascencia** [visgl/deck.gl#6857] [Q&A] [answered]
I would like to create an IconLayer using icons that have a transparent area as part of the icon but are still able to detect picking events. DeckGL does provide a solution which is to set the alphaCutoff to 0.0. [alphaCutoff](https://deck.gl/docs/api-reference/layers/icon-layer#alphacutoff).

The problem I'm encountering is when setting alphaCutoff to 0.0, I'm seeing overlapping icons in the layer "cover up" other icons.
![overalapping_covering](https://user-images.githubusercontent.com/5034592/165114919-5ecbcf98-7d23-467d-9155-4dbf36e53090.png)

I created a codepen example of this behavior based on the DeckGL IconLayer example at: https://codepen.io/mplascencia/pen/jOYRmRL

Is it possible to use alphaCutoff=0.0 for picking events, but still display the icons with the transparency so that icons do not "cover up" each other as in the DeckGL IconLayer example?

![overalapping_transparent](https://user-images.githubusercontent.com/5034592/165117160-9952d92a-fc10-435a-bb46-ecfe91223cdf.png)

Thank you in advance for any help you can provide!

**Comment by Pessimistress** [comment, marked as answer]
Either use a larger [pickingRadius](https://deck.gl/docs/api-reference/core/deck#pickingradius) or render another transparent layer underneath. `alphaCutoff` applies to the rendering to both screen and the picking buffer.

**Reply by mplascencia** [reply]
@Pessimistress When rendering another transparent layer underneath, do you mean there would be a corresponding transparent icon layer for each "real" icon being displayed?  Or, is it a single large transparent layer that fills the entire screen size?  Thanks!

**Reply by Pessimistress** [reply]
You can render any instanced layer (IconLayer, ScatterplotLayer or SimpleMeshLayer) as long as they offer the desirable hit area shape. I assume you still want to pick each object individually, so there must be one geometry for each target.
